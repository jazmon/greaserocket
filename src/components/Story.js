// @flow
import React from 'react';

import Image from 'react-native-image-progress';
import styled from 'styled-components/native';
import type { ThemeType } from 'constants/theme';

import type { Story as StoryType } from 'types';

type Props = {
  story: StoryType,
  style?: ?any,
};

const getTextColor = ({
  theme,
  secondary = false,
}: { theme: ThemeType, secondary?: boolean }) =>
  (secondary ? theme.text.colors.secondary : theme.text.colors.primary);

const Container = styled.View`
  flex: 1;
  flex-direction: row;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
  height: 120;
`;
const ImageContainer = styled.View`
  width: 120;
  align-items: center;
  justify-content: center;
`;
const AuthorImage = styled(Image)`
  width: 100;
  height: 100;
`;
const TextArea = styled.View`
  flex: 1;
  padding: 16;
  flex-direction: column;

`;
const BaseText = styled.Text`
  color: ${props => getTextColor(props)};
  font-size: 16;
`;

const Story = ({ story, style }: Props): ElementType => (
  <Container style={style}>
    <ImageContainer>
      <AuthorImage
        source={{ uri: story.author.profilePictureUrl }}
        resizemode="contain"
      />
    </ImageContainer>
    <TextArea>
      <BaseText secondary>{story.author.name}</BaseText>
      <BaseText>{story.text}</BaseText>
    </TextArea>
  </Container>
);

export default Story;
