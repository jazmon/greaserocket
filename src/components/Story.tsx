// @flow
import * as React from 'react';
import { ViewStyle } from 'react-native';

import Image from 'react-native-image-progress';
import styled from 'styled-components/native';
import { ThemeType } from 'constants/theme';
import defaultPicture from 'assets/default_user.png';

import { Story as StoryType, Maybe } from 'types';

interface Props {
  story: StoryType;
  style?: ViewStyle;
}

interface TextColorProps {
  theme: ThemeType;
  secondary?: boolean;
}

const getTextColor = ({ theme, secondary = false }: TextColorProps): string =>
  secondary ? theme.text.colors.secondary : theme.text.colors.primary;

const Container = styled.View`
  flex: 1;
  flex-direction: row;
  background-color: ${({ theme }: { theme: ThemeType }) => theme.colors.white};
  height: 120px;
`;
const ImageContainer = styled.View`
  width: 120px;
  align-items: center;
  justify-content: center;
`;
const AuthorImage = styled(Image)`
  width: 100px;
  height: 100px;

`;
const TextArea = styled.View`
  flex: 1;
  padding: 16px;
  flex-direction: column;
`;
const BaseText = styled.Text`
  color: ${(props: { secondary?: boolean; theme: ThemeType }) =>
    getTextColor(props)};
  font-size: 16;
`;

const Story = ({ story, style }: Props): JSX.Element =>
  <Container style={style}>
    <ImageContainer>
      <AuthorImage
        source={
          story.author.picture ? { uri: story.author.picture } : defaultPicture
        }
        resizemode="cover"
      />
    </ImageContainer>
    <TextArea>
      <BaseText secondary>
        {story.author.name}
      </BaseText>
      <BaseText>
        {story.content}
      </BaseText>
    </TextArea>
  </Container>;

export default Story;
