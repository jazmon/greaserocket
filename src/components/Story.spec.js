import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Story from './Story';

const story = {
  id: 'foo',
  text: 'foo',
  author: { name: 'foo', profilePictureUrl: null },
};

it('renders correctly', () => {
  const tree = renderer.create(<Story story={story} />);
  expect(tree).toMatchSnapshot();
});
