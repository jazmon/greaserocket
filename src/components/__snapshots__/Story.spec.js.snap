// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 120,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": 120,
        },
        undefined,
      ]
    }
  >
    <Image
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      resizemode="cover"
      source={1}
      style={
        Array [
          Object {
            "height": 100,
            "width": 100,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      secondary={true}
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.54)",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    />
  </View>
</View>
`;
